File: frontend/src/lib/types/inputs.ts
export type InputStatus = "pending" | "answered" | "auto_answered" | "rejected" | "timeout";

export interface JobInput {
  input_id: string;
  job_id: string;
  issue_id: string;
  question: string;
  category: string;
  options: string[] | null;
  status: InputStatus;
  answer: string | null;
  auto_answer: string | null;
  created_at: string;
  answered_at: string | null;
}

export type InputImportance = "blocking" | "auto_answerable";
export type IssueInputStatus = "pending" | "auto_answered" | "answered";

export interface InputResponse {
  id: string;
  issue_id: string;
  job_id: string;
  asking_node: string;
  question: string;
  importance: InputImportance;
  auto_answer: string | null;
  answer: string | null;
  answered_by: string | null;
  status: IssueInputStatus;
  created_at: string;
  answered_at: string | null;
}

File: legion/api/schemas/inputs.py
"""Pydantic request/response models for issue inputs."""

from datetime import datetime

from pydantic import BaseModel, Field


class InputResponse(BaseModel):
    id: str
    issue_id: str
    job_id: str
    asking_node: str
    question: str
    importance: str
    auto_answer: str | None
    answer: str | None
    answered_by: str | None
    status: str
    created_at: datetime
    answered_at: datetime | None


class AnswerInputRequest(BaseModel):
    answer: str = Field(..., min_length=1, max_length=10000)
